### PR TITLE
add a kill command

### PR DESCRIPTION
**Pull request by matthiasclasen**
This sends SIGKILL to a running sandbox process.
The command has completion for running applications.

### CODE REVIEW

**Comment by matthiasclasen** [comment]
This is basically https://github.com/flatpak/flatpak/issues/2180

**Comment by matthiasclasen** [comment]
As alex pointed out in irc, this should probably use more proper container killing technology, like the freezer

File: app/flatpak-builtins-kill.c [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+  instances = flatpak_instance_get_all ();
+
+  for (j = 0; j < instances->len; j++)
+    {
+      FlatpakInstance *instance = (FlatpakInstance *)g_ptr_array_index (instances, j);
+      if (strcmp (app, flatpak_instance_get_app (instance)) == 0)
+        {
+          pid_t pid = flatpak_instance_get_child_pid (instance);
+          kill (pid, SIGKILL);
+          return TRUE;
```
Comment (alexlarsson) [inline app/flatpak-builtins-kill.c]: What if there are multiple instances, i don't think we should return until looking at all.

File: app/flatpak-builtins-kill.c [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+gboolean
+flatpak_builtin_kill (int            argc,
+                      char         **argv,
+                      GCancellable  *cancellable,
+                      GError       **error)
+{
+  g_autoptr(GOptionContext) context = NULL;
+  const char *app;
+
+  context = g_option_context_new (_("NAME - Stop a running application"));
```
Comment (alexlarsson) [inline app/flatpak-builtins-kill.c]: Should we not support specifying an instance id too?

File: app/flatpak-builtins-enter.c [hunk 3]
```diff
@@ -79,7 +80,7 @@ flatpak_builtin_enter (int           argc,
   uid_t uid;
   gid_t gid;
 
-  context = g_option_context_new (_("SANDBOXEDPID [COMMAND [args...]] - Run a command inside a running sandbox"));
+  context = g_option_context_new (_("PID-OR-APP [COMMAND [args...]] - Run a command inside a running sandbox"));
```
Comment (alexlarsson) [inline app/flatpak-builtins-enter.c]: Would be nice to accept an intance id here too.

File: app/flatpak-builtins-kill.c (hunk 1 above)
Comment (matthiasclasen) [reply app/flatpak-builtins-kill.c]: Currently, we look up what to kill by the (unique) application ID, so I don't think there can be multiple matches, but we can do that, to be future-proof.

File: app/flatpak-builtins-kill.c (hunk 2 above)
Comment (matthiasclasen) [reply app/flatpak-builtins-kill.c]: would be nice. I just wanted to start simple

File: app/flatpak-builtins-enter.c (hunk 3 above)
Comment (matthiasclasen) [reply app/flatpak-builtins-enter.c]: I'll look into that

File: app/flatpak-builtins-kill.c (hunk 1 above)
Comment (alexlarsson) [reply app/flatpak-builtins-kill.c]: The application id is not uniqe though? You can easily start two copies of the same app (unless it does uniqeness itself)

Comment (matthiasclasen) [reply app/flatpak-builtins-kill.c]: I was assuming the app takes a bus name...

**Comment by alexlarsson** [comment]
@rh-atomic-bot r+

**Comment by alexlarsson** [comment]
@rh-atomic-bot r+

**Pull request closed without merging**